### PR TITLE
PHP: removed built-in type real

### PR DESCRIPTION
**Pull request by troizet**
The main purpose of this PR is to fix the situation where NetBeans generates a `real` type and uses it as a native return type.

In this PR:
 - deleted the built-in type `real` (php does not have a built-in `real` type: https://www.php.net/manual/en/language.types.intro.php Also see the warning under "Scalar Types": https://www.php.net/manual/en/language.types.declarations.php)
  - fixed broken tests
  - new tests added
  - fixes generation of return type `real` instead of `float` for a method when overriding a base class method with an autocomplete
  - fixes generation of return type `real` instead of `float` for a function in phpDoc

This PR is similar to #5294.

### CODE REVIEW

**Comment by troizet** [comment]
@mbien Can I ask for the tests to be restarted? I don't have the ability to set the label before creating the PR.

File: php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/CastExpression.java
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/CastExpression.java:30]: I don't think we should change this because this is an API.

@tmysik What do you think?

File: php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/Scalar.java
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/Scalar.java:32]: I don't think we should change this because this is an API.

Comment (tmysik) [reply php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/Scalar.java:32]: Right, we cannot do this. If you really want to do that, the steps should be like:
- in release X
   - deprecate the `REAL` type + document all the details (why deprecated and when it will be removed)
   - add new `FLOAT` type
   - change the code under our control to start using the `FLOAT` type
- in release X + Y (Y is quite a big number, we cannot do it in the near future, we need to give time to our API consumers - plugins - to change their code):
   - remove the `REAL` type

File: php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/CastExpression.java
Comment (tmysik) [reply php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/CastExpression.java:30]: Yes, see [my comment](https://github.com/apache/netbeans/pull/7387#discussion_r1608051235).

**Comment by tmysik** [comment]
_Note:_ If we already did something similar in #5294, then it is unfortunate, as I explained above. Ideally, we should revert the API change and follow the steps I described.

OTOH, not sure how many plugins exist that are affected by this change and therefore can be broken. Also, the fix should be easy, but we must avoid of such API changes, definitely.

CC @junichi11

File: php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/Scalar.java
Comment (troizet) [reply php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/Scalar.java]: @tmysik Thank you very much for the explanation. 

Updated the PR:
- removed the `real` type removal in the public api.
- increased the minor version of the module.

I hope I did everything correctly.

**Comment by troizet** [comment]
> _Note:_ If we already did something similar in #5294, then it is unfortunate, as I explained above. Ideally, we should revert the API change and follow the steps I described.

By similarity to #5294, I was referring to the purpose of that PR.
If I understood it correctly, there were no changes to the public api in that PR.

File: php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/CastExpression.java
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/CastExpression.java:36]: We should not only add `FLOAT` but also deprecate `REAL`, no? Ideally with a comment that it will be removed in the future, to follow PHP type system.

File: php/php.editor/src/org/netbeans/modules/php/editor/typinghooks/PhpCommentGenerator.java
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/typinghooks/PhpCommentGenerator.java:544]: Is it safe to remove this line? Should not we remove it in the future, together once the `REAL` type is removed?

CC @junichi11

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/PHP54UnhandledError.java
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/verification/PHP54UnhandledError.java:159]: @junichi11 Again, please, verify that we can simply replace `REAL` -> `FLOAT` and not keep the both types. Thank you.

**Review by tmysik — APPROVED** [review]
Approved, but please wait for the approval by @junichi11. Thank you!

File: php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/CastExpression.java
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/CastExpression.java:36]: Agreed. 

e.g.
```java
        /**
         * @deprecated will be removed in the future, to follow PHP type system,
         * use {@link Type#FLOAT} instead
         */
        REAL, // 'real'
```

File: php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/Scalar.java
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/Scalar.java:36]: The same as above.

File: php/php.editor/src/org/netbeans/modules/php/editor/typinghooks/PhpCommentGenerator.java
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/typinghooks/PhpCommentGenerator.java:544]: It should be OK (I assume all `REAL` types were replaced with `FLOAT` in the parser).
But I would leave it, just in case.

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/PHP54UnhandledError.java
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/verification/PHP54UnhandledError.java:159]: It should be OK (I assume all `REAL` types were replaced with `FLOAT` in the parser).
But I would keep both types, just in case.

Comment (tmysik) [reply php/php.editor/src/org/netbeans/modules/php/editor/verification/PHP54UnhandledError.java:159]: The only use case would be if we had it in the index, right? Otherwise, we should be safe.

But as we agreed, I believe that keeping both types should be super-safe. Once we remove the `REAL` type, it will be very easy to delete all its usages.

@troizet, could you, please, update this PR? Thanks for understanding, and sorry for any inconveniences, we are trying to be really safe here.

**Review by tmysik — CHANGES_REQUESTED** [review]
See the comments - please, keep the `REAL` type and add the `FLOAT` one. Later on, we will remove the `REAL` type with all its usages.

CC @junichi11

**Comment by troizet** [comment]
Made the suggested corrections.

**Review by tmysik — APPROVED** [review]
Looks good to me, thanks a lot for this PR!

**Review by junichi11 — APPROVED** [review]
Thanks for your work!

**Comment by troizet** [comment]
@junichi11 Forgot to set milestone for this PR, or is it optional? Thank you!

File: php/php.editor/test/unit/src/org/netbeans/modules/php/editor/completion/PHPCCDocumentationTest.java
Comment (mbien) [inline php/php.editor/test/unit/src/org/netbeans/modules/php/editor/completion/PHPCCDocumentationTest.java:659]: this test started failing recently on linux. Could this be some kind of race condition where it is sometimes returning a different completion item?

it works fine on windows.

Comment (junichi11) [reply php/php.editor/test/unit/src/org/netbeans/modules/php/editor/completion/PHPCCDocumentationTest.java:659]: I also noticed it. I'll fix it ASAP. Thanks.

Comment (junichi11) [reply php/php.editor/test/unit/src/org/netbeans/modules/php/editor/completion/PHPCCDocumentationTest.java:659]: It should be fixed in https://github.com/apache/netbeans/pull/7553. Sorry for the inconvenience.

Comment (tmysik) [reply php/php.editor/test/unit/src/org/netbeans/modules/php/editor/completion/PHPCCDocumentationTest.java:659]: Thanks a lot, @junichi11!

Comment (mbien) [reply php/php.editor/test/unit/src/org/netbeans/modules/php/editor/completion/PHPCCDocumentationTest.java:659]: this was a bit of a dejavu for me but I can't remember where this happened before (groovy? javascript? php?)

what do you think about this change:
```diff
diff --git a/ide/csl.api/test/unit/src/org/netbeans/modules/csl/api/test/CslTestBase.java b/ide/csl.api/test/unit/src/org/netbeans/modules/csl/api/test/CslTestBase.java
index 4b21823..8eff017 100644
--- a/ide/csl.api/test/unit/src/org/netbeans/modules/csl/api/test/CslTestBase.java
+++ b/ide/csl.api/test/unit/src/org/netbeans/modules/csl/api/test/CslTestBase.java
@@ -107,6 +107,7 @@
 import java.util.logging.Handler;
 import java.util.logging.Level;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 import javax.swing.JEditorPane;
 import javax.swing.event.DocumentEvent;
 import javax.swing.event.DocumentEvent.ElementChange;
@@ -182,6 +183,7 @@
 import org.openide.util.Pair;
 import org.openide.util.lookup.Lookups;
 import org.openide.util.test.MockLookup;
+
 import static org.openide.util.test.MockLookup.setLookup;
 
 /**
@@ -3145,13 +3147,12 @@
                 CodeCompletionResult completionResult = cc.complete(context);
                 List<CompletionProposal> proposals = completionResult.getItems();
 
-                CompletionProposal match = null;
-                for (CompletionProposal proposal : proposals) {
-                    if (proposal.getName().startsWith(itemPrefix)) {
-                        match = proposal;
-                        break;
-                    }
-                }
+                List<CompletionProposal> matched = proposals.stream()
+                                                            .filter(p -> p.getName().startsWith(itemPrefix))
+                                                            .collect(Collectors.toList());
+                assertEquals("more than one item matched the prefix", 1, matched.size());
+                CompletionProposal match = matched.get(0);
+
                 assertNotNull(match);
                 assertNotNull(match.getElement());
 

```
this would avoid prevent this situation to re-occur, however, this would also cause many other test failures right now.

Comment (troizet) [reply php/php.editor/test/unit/src/org/netbeans/modules/php/editor/completion/PHPCCDocumentationTest.java:659]: @mbien @junichi11 I apologize for the problematic tests. Couldn't this also be related to the #6066?

Comment (junichi11) [reply php/php.editor/test/unit/src/org/netbeans/modules/php/editor/completion/PHPCCDocumentationTest.java:659]: @mbien `matched.size()` may not be `1`.

Comment (junichi11) [reply php/php.editor/test/unit/src/org/netbeans/modules/php/editor/completion/PHPCCDocumentationTest.java:659]: @troizet Maybe, It is not related to this.

Comment (mbien) [reply php/php.editor/test/unit/src/org/netbeans/modules/php/editor/completion/PHPCCDocumentationTest.java:659]: > @mbien `matched.size()` may not be `1`.

in that case it needs to be sorted so that it doesn't randomly fail

Comment (junichi11) [reply php/php.editor/test/unit/src/org/netbeans/modules/php/editor/completion/PHPCCDocumentationTest.java:659]: Yes, the cause is it, I think.

**Pull request merged**
